operator/v1/types_ingress.go

```diff
@@ -1004,14 +1001,19 @@ type IngressControllerHTTPHeaders struct {
 
 // IngressControllerHTTPHeaderBuffer specifies the size of the
 // per-connection HTTP header buffers.
-type IngressControllerHTTPHeaderBuffer struct {
+type IngressControllerTuningOptions struct {
 	// headerBufferBytes describes how much memory should be reserved
 	// (in bytes) for IngressController connection sessions.
 	// Note that this value must be at least 16384 if HTTP/2 is
 	// enabled for the IngressController (https://tools.ietf.org/html/rfc7540).
 	// If this field is empty, the IngressController will use a default value
 	// of 32768 bytes.
 	//
+	// Setting this field is generally not recommended as header buffer
```

Comment:
This warning essentially applies to all of the fields in `IngressControllerTuningOptions`, right? Maybe we should move this warning to the `TuningOptions` godoc (and revise it accordingly).

Reply:
> This warning essentially applies to all of the fields in `IngressControllerTuningOptions`, right? Maybe we should move this warning to the `TuningOptions` godoc (and revise it accordingly).

Duplicating it makes sense. But I like having it on each field with the impact described individually.

Reply:
I agree with duplicating it. I've added this type of warning to both `spec.tuningOptions` and `spec.tuningOptions.threadCount`. I think a warning could go on `spec.tuningOptions.HeaderBufferMaxRewriteBytes` too, but I'm not sure how to word it.

Reply:
`HeaderBufferMaxRewriteBytes` and `HeaderBufferBytes` both have the potential to either break routes or cause HAProxy to use too much memory. I think it would be sufficient to re-use the warning here for `HeaderBufferMaxRewriteBytes` (May want to change `header buffer values` to be the name of the corresponding field in both warnings.

Reply:
Fixed!